benchtop/src/nomt.rs

```diff
@@ -61,14 +60,71 @@ impl NomtDB {
         self.nomt.commit_and_prove(session, actual_access).unwrap();
     }
 
+    // note: this is only intended to be used with workloads which are disjoint, i.e. no workload
+    // writes a key which another workload reads. re-implementing BlockSTM or other OCC methods are
+    // beyond the scope of benchtop.
+    pub fn parallel_execute(
+        &self,
+        mut timer: Option<&mut Timer>,
+        thread_pool: &rayon::ThreadPool,
+        workloads: &mut [&mut dyn Workload],
+    ) {
+        let _timer_guard_total = timer.as_mut().map(|t| t.record_span("workload"));
+
+        let session = self.nomt.begin_session();
+        let mut results: Vec<Option<_>> = (0..workloads.len()).map(|_| None).collect();
+
+        let use_timer = timer.is_some();
+        thread_pool.in_place_scope(|scope| {
+            for (workload, result) in workloads.into_iter().zip(results.iter_mut()) {
+                let session = &session;
+                scope.spawn(move |_| {
+                    let mut workload_timer = if use_timer {
+                        Some(Timer::new(String::new()))
+                    } else {
+                        None
+                    };
+                    let mut transaction = Tx {
+                        session,
+                        access: FxHashMap::default(),
+                        timer: workload_timer.as_mut(),
+                    };
+                    workload.run_step(&mut transaction);
+                    *result = Some((transaction.access, workload_timer.map(|t| t.freeze())));
+                })
+            }
+        });
+
+        // absorb instrumented times from workload timers.
+        for workload_timer in results
+            .iter_mut()
+            .flatten()
+            .map(|(_, ref mut workload_timer)| workload_timer)
```

Comment:
just style, this map could be avoided and something like `for (_, ref mut workload_timer) in results.iter_mut().flatten() {` should work